File: android/app/src/main/java/com/acmegames/vibearcade/core/networking/ApiConfiguration.kt
package com.acmegames.vibearcade.core.networking

object ApiConfiguration {
    val BASE_URL = Environment.getEffectiveBaseUrl()
    val WEBSOCKET_URL = Environment.getEffectiveWebSocketUrl()

    // Endpoints
    object Endpoints {
        const val REGISTER = "auth/register"
        const val LOGIN = "auth/login"
        const val REFRESH_TOKEN = "auth/refresh"
        const val PROFILE = "user/profile"
        const val GAMES = "games"
        const val CREATE_GAME = "games"

        fun joinGame(gameId: String) = "games/$gameId/join"
        fun gameDetails(gameId: String) = "games/$gameId"
        fun makeMove(gameId: String) = "games/$gameId/move"
    }

    // Headers
    const val CONTENT_TYPE = "application/json"
    const val AUTHORIZATION = "Authorization"
    const val BEARER = "Bearer "
}

File: ios/VibeArcade/Core/Networking/APIConfiguration.swift
import Foundation

struct APIConfiguration {
    static let shared = APIConfiguration()

    private init() {}

    // MARK: - Base Configuration
    let baseURL = Environment.shared.baseURL
    let websocketURL = Environment.shared.websocketURL

    // MARK: - Endpoints
    enum Endpoint {
        case register
        case login
        case refreshToken
        case profile
        case games
        case createGame
        case joinGame(String)
        case gameDetails(String)
        case makeMove(String)

        var path: String {
            switch self {
            case .register:
                return "/auth/register"
            case .login:
                return "/auth/login"
            case .refreshToken:
                return "/auth/refresh"
            case .profile:
                return "/user/profile"
            case .games:
                return "/games"
            case .createGame:
                return "/games"
            case .joinGame(let gameId):
                return "/games/\(gameId)/join"
            case .gameDetails(let gameId):
                return "/games/\(gameId)"
            case .makeMove(let gameId):
                return "/games/\(gameId)/move"
            }
        }

        var url: URL {
            return URL(string: APIConfiguration.shared.baseURL + path)!
        }
    }

    // MARK: - HTTP Methods
    enum HTTPMethod: String {
        case GET = "GET"
        case POST = "POST"
        case PUT = "PUT"
        case DELETE = "DELETE"
    }

    // MARK: - Headers
    var defaultHeaders: [String: String] {
        return [
            "Content-Type": "application/json",
            "Accept": "application/json"
        ]
    }

    func authorizedHeaders(token: String) -> [String: String] {
        var headers = defaultHeaders
        headers["Authorization"] = "Bearer \(token)"
        return headers
    }
}